DataMigration/src/main/java/org/sagebionetoworks/dian/datamigration/SynapseUtil.java

```
@@ -0,0 +1,970 @@
+                    String.join(", ", failedToParseList));
+        }
+
+        return map;
+    }
+
+    private static List<String> jsonFilePathList(@NonNull File extractedFolder) {
+        List<String> retVal = new ArrayList<>();
+        File[] fileList = extractedFolder.listFiles();
+        if (fileList == null || fileList.length <= 0) {
+            return retVal;
+        }
+
+        List<String> ignoredFileList = new ArrayList<>();
+
+        // Extracted ZIPs are one
+        // Move through the folder structure to get to the JSON files
+        for (File file: fileList) {
+            // Iterate through the folder/file structure of the unzipped test session content
+            if (file.isDirectory()) {
+                String[] filePathList = new String[0];
+                String[] expectedDirPathList = file.list();
+                if (expectedDirPathList != null) {
+                    filePathList = expectedDirPathList;
+                }
+                // Iterate through the list of folders of days unzipped
+                for (String filename : filePathList) {
+                    // Ignore everything but JSON files
+                    if (filename.endsWith(".json")) {
+                        retVal.add(file.getAbsolutePath() + File.separator + filename);
+                    } else {
+                        ignoredFileList.add(filename);
+                    }
+                }
+            }
+        }
+
+        if (!ignoredFileList.isEmpty()) {
+            System.out.println("Ignored parsing file(s) " +
+                    String.join(", ", ignoredFileList));
+        }
+
+        return retVal;
+    }
+
+    public static @Nullable File createFolderIfNecessary(String folderName) {
+        File folder = new File(folderName);
+        if (!folder.exists() || !folder.isDirectory()) {
+            if (!folder.mkdir()) {
+                System.out.println("Could not create folder " + folderName);
```

shouldn't this raise an exception instead of returning null?